DuggaSys/sectioned.js

```diff
@@ -2279,8 +2297,19 @@ function drawSwimlanes() {
           textcol = `url("#fadeTextRed")`;
         }
 
-        str += `<rect opacity='0.7' x='${(startday * daywidth)}' y='${(weeky)}' width='${(duggalength * daywidth)}' height='${weekheight}' fill='${fillcol}' />`;
-        str += `<text x='${(12)}' y='${(weeky + 18)}' font-family='Arial' font-size='12px' fill='${textcol}' text-anchor='left'> <title> ${entry.text} </title>${entry.text}</text>`;
+        if(storedTheme.includes("blackTheme.css")){ //If theme is black then dugga progress should be more pigmented
+          str += `<rect opacity='0.9' x='${(startday * daywidth)}' y='${(weeky)}' width='${(duggalength * daywidth)}' height='${weekheight}' fill='${fillcol}' />`;
+        }else{ //If theme is white then week numbers in the course schedule should be less pigmented
+          str += `<rect opacity='0.7' x='${(startday * daywidth)}' y='${(weeky)}' width='${(duggalength * daywidth)}' height='${weekheight}' fill='${fillcol}' />`;
+        }
+
+        //Changes text colors of assignments/duggor in course SVG (course schedule with deadlines)
+        if(storedTheme.includes("blackTheme.css")){ //If theme is black then duggor-text in the course schedule should be white
+          str += `<text x='${(12)}' y='${(weeky + 18)}' font-family='Arial' font-size='12px' fill='#ffffff' text-anchor='left'> <title> ${entry.text} </title>${entry.text}</text>`;
+        }else{ //If theme is white then duggor-text in the course schedule should be black
+          str += `<text x='${(12)}' y='${(weeky + 18)}' font-family='Arial' font-size='12px' fill='${textcol}' text-anchor='left'> <title> ${entry.text} </title>${entry.text}</text>`;
+        }
+        
```

Comment:
This could probably also be handled automatically if they have the correct parent css class.